siatest/consensus/consensus_test.go

```diff
@@ -56,76 +57,152 @@ func TestConsensusBlocksIDGet(t *testing.T) {
-	}
-	if block.ParentID == zeroID {
-		t.Fatal("ParentID wasn't set correctly")
-	}
-	if block.Timestamp == types.Timestamp(0) {
-		t.Fatal("Timestamp wasn't set correctly")
-	}
-	if len(block.MinerPayouts) == 0 {
-		t.Fatal("Block has no miner payouts")
-	}
-	if len(block.Transactions) == 0 {
-		t.Fatal("Block doesn't have any transactions even though it should")
-	}
+	for i = 0; i <= endBlock.Height; i++ {
+		cbhg, err := testNode.ConsensusBlocksHeightGet(i)
+		if err != nil {
+			t.Fatal("Failed to retrieve block by height:", err)
+		}
+		cbig, err := testNode.ConsensusBlocksIDGet(cbhg.ID)
+		if err != nil {
+			t.Fatal("Failed to retrieve block by ID:", err)
+		}
+		// Confirm blocks received by both endpoints are the same
+		if !reflect.DeepEqual(cbhg, cbig) {
+			t.Fatal("Blocks not equal")
+		}
+		// Confirm Fields were set properly
+		// Ignore ParentID and MinerPayouts for genisis block
+		if cbig.ParentID == zeroID && i != 0 {
+			t.Fatal("ParentID wasn't set correctly")
+		}
+		if len(cbig.MinerPayouts) == 0 && i != 0 {
+			t.Fatal("Block has no miner payouts")
+		}
+		if cbig.Timestamp == types.Timestamp(0) {
+			t.Fatal("Timestamp wasn't set correctly")
+		}
+		if len(cbig.Transactions) == 0 {
+			t.Fatal("Block doesn't have any transactions even though it should")
+		}
 
-	// Get same block by height
-	block2, err := testNode.ConsensusBlocksHeightGet(cg.Height)
-	if err != nil {
-		t.Fatal("Failed to retrieve block", err)
-	}
-	// block and block2 should be the same
-	if block.ID != block2.ID {
-		t.Fatal("BlockID wasn't set correctly")
-	}
-	if block.Height != block2.Height {
-		t.Fatal("BlockID wasn't set correctly")
-	}
-	if block.ParentID != block2.ParentID {
-		t.Fatal("ParentIDs don't match")
-	}
-	if block.Timestamp != block2.Timestamp {
-		t.Fatal("Timestamps don't match")
-	}
-	if len(block.MinerPayouts) != len(block2.MinerPayouts) {
-		t.Fatal("MinerPayouts don't match")
-	}
-	if len(block.Transactions) != len(block2.Transactions) {
-		t.Fatal("Transactions don't match")
+		// Verify IDs
+		for _, tx := range cbhg.Transactions {
+			// Building transaction of type Transaction to use as
+			// comparison for ID creatation
+			txn := types.Transaction{
+				SiacoinInputs:         tx.SiacoinInputs,
+				FileContractRevisions: tx.FileContractRevisions,
+				StorageProofs:         tx.StorageProofs,
+				SiafundInputs:         tx.SiafundInputs,
+				MinerFees:             tx.MinerFees,
+				ArbitraryData:         tx.ArbitraryData,
+				TransactionSignatures: tx.TransactionSignatures,
+			}
+			if len(tx.SiacoinOutputs) != 0 {
+				// Finish building Transaction
+				for _, sco := range tx.SiacoinOutputs {
+					txn.SiacoinOutputs = append(txn.SiacoinOutputs, types.SiacoinOutput{
+						Value:      sco.Value,
+						UnlockHash: sco.UnlockHash,
+					})
+				}
+				// Verify SiacoinOutput IDs
+				for i, sco := range tx.SiacoinOutputs {
+					if sco.ID != txn.SiacoinOutputID(uint64(i)) {
```

Comment:
I think you actually have to finish building the whole transaction before you can compute any ids. 
Iirc `SiacoinOutputID`, `SiafundOutputID`, etc use all the fields of the transaction but the signatures.